pkg/env/env.go

```diff
@@ -116,6 +117,18 @@ func newTestEnvWithParallel() *testEnv {
 	}
 }
 
+// newChildTestEnv returns a child testEnv based on the one passed as an argument.
+// The child env inherits the context and actions from the parent and
+// creates a deep copy of the config so that it can be mutated without
+// affecting the parent's.
+func newChildTestEnv(e *testEnv) *testEnv {
+	return &testEnv{
+		ctx:     e.ctx,
+		cfg:     e.deepCopyConfig(),
+		actions: e.actions,
```

Comment:
Should we create a new slice here as we do in [WithContext](https://github.com/kubernetes-sigs/e2e-framework/blob/0ba3d3baf4c53aa9bc90f641a9f603ec6b295e01/pkg/env/env.go#L127), to avoid possible race conditions on that too.

Reply:
sure, it's not a perfect copy though: since the underlying `funcs`, `featureFuncs` and `testFuncs` fields of each action are also slices, they are still shared. But I don't think there is any way in the current code to alter those slices after creation so it should not cause any issue even if we just copy the `[]action` slice.

Would you prefer a deep copy of the actions as well to be safe?

Reply:
I've copied the `actions` like it is done in `WithContext` for now and I've also created a child context so that none of the fields of the testEnv are shared